### PR TITLE
FIX Handle string null CallToActionLink in database

### PR DESCRIPTION
**Pull request by emteknetnz**
Fixes https://github.com/silverstripe/silverstripe-elemental-bannerblock/issues/30

See comment https://github.com/silverstripe/silverstripe-elemental-bannerblock/issues/30#issuecomment-555460856

### CODE REVIEW

File: src/Block/BannerBlock.php [hunk 1]
```diff
@@ -52,11 +52,25 @@ public function getCMSFields()
...
         return $this->decodeLinkData($this->getField('CallToActionLink'));
     }
 
+    /**
+     * Alias of getCallToActionLink()
+     * Originally this was the only function, however it was not being called by PHP
+     * i.e. $bannerBlock->CallToActionLink
+     *
+     * This function is only being kept for semver as removing it would be a breaking change
+     *
```
Comment (ScopeyNZ) [inline src/Block/BannerBlock.php:66]:
```suggestion
     *
     * @deprecated 3.0.0
```

Comment (emteknetnz) [reply src/Block/BannerBlock.php:66]: My original solution caused a regression which showed up in a failed behat test.  I've redone my solution.

File: src/Block/BannerBlock.php [hunk 2]
```diff
@@ -47,6 +47,19 @@ public function getCMSFields()
...
+     * For the backend, e.g. $block->CallToActionLink, return the database column which is
+     * a json encoded string.  If there is old data where null was mistakenly saved as a string 'null'
+     * then convert that to null
+     *
+     * @return string|null
+     */
+    public function getCallToActionLink()
+    {
+        $callToActionLink = $this->getField('CallToActionLink');
+        return $callToActionLink === 'null' ? null: $callToActionLink;
```
Comment (michalkleiner) [inline src/Block/BannerBlock.php]: Add a space after the `:` in the ternary?

Comment (emteknetnz) [reply src/Block/BannerBlock.php]: Yes ternary, surprised phpcs didn't pick it up.  Have fixed.

File: src/Block/BannerBlock.php [hunk 3]
```diff
@@ -47,6 +47,19 @@ public function getCMSFields()
         return parent::getCMSFields();
     }
 
+    /**
+     * For the backend, e.g. $block->CallToActionLink, return the database column which is
+     * a json encoded string.  If there is old data where null was mistakenly saved as a string 'null'
+     * then convert that to null
```
Comment (maxime-rainville) [inline src/Block/BannerBlock.php]: I would put this comment in the body of the method. Developers calling this method from another class don't need to understand the underlying purpose.

```suggestion
     * Accessor for `CallToActionLink`.
```

**Review by maxime-rainville — APPROVED** [review]
Tested it locally. Works great. 

Left a suggestion for a comment. Do with it as you please.

File: src/Block/BannerBlock.php (hunk 3 above)
Comment (emteknetnz) [reply src/Block/BannerBlock.php]: Good suggestion, have updated comments

**Pull request merged**